packages/calypso-e2e/src/hooks.ts

```
@@ -64,24 +64,26 @@ export const setupHooks = ( callback: ( { page }: { page: Page } ) => void ): vo
 			await page.screenshot( { path: fileName } );
 		}
 		// Close the browser. This needs to be called before trying to access
-		// the video recording
+		// the video recording.
 		await page.close();
-		// Save video
+
 		if ( __CURRENT_TEST_FAILED__ ) {
-			const original = await ( page.video() as Video ).path();
+			// const original = await ( page.video() as Video ).path();
 			const destination = path.join(
 				artifactPath,
 				'screenshots',
 				`${ getTestNameWithTime( testName ) }.webm`
 			);
 			try {
-				await rename( original, destination );
+				// Save the failing test case with a specific name.
+				await page.video()?.saveAs( destination );
 			} catch ( err ) {
-				console.error( 'Failed to rename video of failing test case.' );
+				console.error( 'Failed to save video of failing test case.' );
```

While you are in this area, I think we can improve this and re-throw the error (or at the very least, log it) to avoid missing information

---

It is __possible__ that Playwright guarantees that a video will be saved at the path specified. [Documentation here](https://playwright.dev/docs/api/class-video).

The documentation for `saveAs` does not explicitly use the term "guarantee" but I would like to find out of this is an assumption that can be made. If it is guaranteed, then the `try/catch` is no longer necessary.

I'll follow up with the maintainers on this question.

---

Asked a question with the maintainers on whether all methods in the Video class have guarantees: https://github.com/microsoft/playwright/issues/9324

---

I will follow up on responses (whatever they may be) in a separate task/PR, as currently we are lacking video recording of failed test cases.

---

As discussed below with @dpasque, I will merge this change for now as currently failed runs have no video recorded.

The intention is to look at refactoring how the video and screenshot capture happens, taking into account the possibility of multiple contexts. Filed https://github.com/Automattic/wp-calypso/issues/56805 for this.